### PR TITLE
Ammp 1199 improve sdmo driver

### PR DESCRIPTION
**Pull request by judruizc**
Hey @svet-b I was saving some of the SDMO data with a hidden Kilo (kW, kVA).  Based on the data evaluated by the product team (specially Egon and Jonathan), they assumed the units to be without the Kilo (as they should), and after analysis they found the right multiplier to be 100. 

Apparently the SDMO people do report in _kW_, but they use some notation "Power reading precision" which might be why the right multiplier is not 1000 but 100

For the **Circutor**, I saw some dataloggers were failing to load the driver because of the HEX. So I changed them to decimal. 

Screenshot 2021-01-12 at 15 32 11

Screenshot 2021-01-12 at 15 31 25

### CODE REVIEW

**Comment by svet-b** [comment]
> For the Circutor, I saw some dataloggers were failing to load the driver because of the HEX. So I changed them to decimal.
This doesn't really make sense to me...what's the error you are seeing? Unless the driver is invalid JSON or has invalid characters, there shouldn't be an error with loading it.

File: drivers/circutor_cdp_duo.json [hunk 1]
```diff
@@ -3,28 +3,28 @@
...
-    "grid_in_E": {"register": "0x82", "description": "Active three-phase energy consumed from the grid", "unit": "Wh", "words": 2, "datatype": "uint32", "multiplier": 1},
-    "pv_E_dc": {"register": "0x84", "description": "Three-phase active photovoltaic energy", "unit": "Wh", "words": 2, "datatype": "uint32", "multiplier": 1},
-    "feed_in_E": {"register": "0x86", "description": "Active three-phase energy injected into the grid", "unit": "Wh", "words": 2, "datatype": "uint32", "multiplier": 1},
-    "grid_out_freq": {"register": "0x96", "description": "Frequency load", "unit": "Hz", "words": 2, "datatype": "uint32", "multiplier": 0.1},
-    "grid_in_freq": {"register": "0x14C", "description": "Frequency grid", "unit": "Hz", "words": 2, "datatype": "uint32", "multiplier": 0.1},
-    "pv_freq": {"register": "0x1B0", "description": "Frequency pv", "unit": "Hz", "words": 2, "datatype": "uint32", "multiplier": 0.1}
+    "grid_in_P_L1": {"register": "100", "description": "L1 Active power of the Grid (sign indicates consumed or generated)", "unit": "W", "words": 2, "datatype": "int32", "multiplier": 1},
```
Comment (svet-b) [inline drivers/circutor_cdp_duo.json:6]: If you do want to convert to decimal registers, then they should not be quoted - i.e. numbers and not strings. (but per my other comment, I'm not sure that's the issue here)

File: drivers/sdmo_apm303.json [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
-    "P_L2": {"description": "Generator L2 live power", "register": 13, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kW", "typecast": "float"},
-    "P_L3": {"description": "Generator L3 live power", "register": 14, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kW", "typecast": "float"},
-    "S": {"description": "Generator total effective power", "register": 10, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kVA", "typecast": "float"},
-    "S_L1": {"description": "Generator L1 effective power", "register": 15, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kVA", "typecast": "float"},
-    "S_L2": {"description": "Generator L2 effective power", "register": 16, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kVA", "typecast": "float"},
-    "S_L3": {"description": "Generator L3 effective power", "register": 17, "words": 1, "datatype": "int16", "multiplier": 1, "unit": "kVA", "typecast": "float"},
+    "P": {"description": "Generator total watts", "register": 9, "words": 1, "datatype": "int16", "multiplier": 100, "unit": "W", "typecast": "float"},
```
Comment (svet-b) [inline drivers/sdmo_apm303.json:7]: These look good!

**Comment by judruizc** [comment]
just tested with couple of dataloggers, it's loading the circutor driver fine 😅. Maybe I saw something wrong. Let's skip it.

Should I revert the changes or can you cherry pick the sdmo alone ?
thanks !

**Comment by svet-b** [comment]
> just tested with couple of dataloggers, it's loading the circutor driver fine 😅. Maybe I saw something wrong. Let's skip it.

Hm ok. Do let me know if you see something strange though!

> Should I revert the changes or can you cherry pick the sdmo alone ?

Since the changes to both files are in a single commit I don't think it's possible to cherry pick in the strict sense of the word. As an alternative, I've reverted the changes to the Circutor driver specifically by running
```
git checkout HEAD~1 -- drivers/circutor_cdp_duo.json
```
and then committing the result as a new change on top of your commit (which you can see above)

**Pull request merged**